book/sources/exex/tracking-state/src/bin/2.rs

```diff
@@ -0,0 +1,74 @@
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{ready, Context, Poll},
+};
+
+use alloy_primitives::BlockNumber;
+use futures_util::StreamExt;
+use reth::api::FullNodeComponents;
+use reth_exex::{ExExContext, ExExEvent};
+use reth_node_ethereum::EthereumNode;
+use reth_tracing::tracing::info;
+
+struct MyExEx<Node: FullNodeComponents> {
+    ctx: ExExContext<Node>,
+    /// First block that was committed since the start of the ExEx.
+    first_block: Option<BlockNumber>,
+    /// Total number of transactions committed.
+    transactions: u64,
+}
+
+impl<Node: FullNodeComponents> MyExEx<Node> {
+    fn new(ctx: ExExContext<Node>) -> Self {
+        Self { ctx, first_block: None, transactions: 0 }
+    }
+}
+
+impl<Node: FullNodeComponents> Future for MyExEx<Node> {
+    type Output = eyre::Result<()>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let this = self.get_mut();
+
+        while let Some(result) = ready!(this.ctx.notifications.poll_next_unpin(cx)) {
+            let notification = result?;
+            if let Some(reverted_chain) = notification.reverted_chain() {
+                this.transactions = this.transactions.saturating_sub(
+                    reverted_chain.blocks_iter().map(|b| b.body.transactions.len() as u64).sum(),
```

Comment:
This line was giving an error since reth primitives `BlockBody` doesn't have a method called len(), so I assumed the intention was the number of the transactions. The original example was:

https://github.com/paradigmxyz/reth/blob/7c818c1021abd343a71ccabbc461724b8f68ffe3/book/developers/exex/tracking-state.md?plain=1#L137-L142

Reply:
yep that's right